docs/csharp/methods.md

```diff
@@ -1,27 +1,295 @@
-ms.technology: devlang-csharp
+ms.prod: .net-core
+ms.technology: .net-core-technologies
 ms.devlang: csharp
-ms.assetid: b6a0539a-8ce5-4da7-adcf-44be345a2714
+ms.assetid: 577a8527-1081-4b36-9b9e-0685b6553c6e
 ---
+# Methods #
 
-# 🔧 Methods
-
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/core-docs/issues/488) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md).
->
+A method is a code block that contains a series of statements. A program causes the statements to be executed by calling the method and specifying any required method arguments. In C#, every executed instruction is performed in the context of a method. The `Main` method is the entry point for every C# application and it is called by the common language runtime (CLR) when the program is started.
+
+> [!NOTE]
+> This topic discusses named methods. For information about anonymous functions, see [Anonymous Functions](https://msdn.microsoft.com/en-us/library/bb882516.aspx).
+
+This topic contains the following sections:
+
+- [Method signatures](#signatures)
+- [Method invocation](#invocation)
+- [Inherited and overridden methods](#inherited)
+- [Passing parameters](#passing)
+  - [Passing parameters by value](#byval)
+  - [Passing parameters by reference](#byref)
+  - [Parameter arrays](#paramarray)
+- [Optional parameters and arguments](#optional)
+- [Return values](#return)
+- [Extension methods](#extension)
+- [Async Methods](#async)
+- [Expression body definitions](#expr)
+- [Iterators](#iterators)
+
+## <a name="signatures" /a> Method signatures ##
+
+Methods are declared in a `class` or `struct` by specifying:
+
+- An access level, such as `public` or `private`.
+- Optional modifiers such as `abstract` or `sealed`.
+- The return value, or `void` if the method has none.
+- The method name.
+- Any method parameters. Method parameters are enclosed in parentheses and are separated by commas. Empty parentheses indicate that the method requires no parameters.
+
+These parts together form the method signature.
+
+> [!NOTE]
+> A return type of a method is not part of the signature of the method for the purposes of method overloading. However, it is part of the signature of the method when determining the compatibility between a delegate and the method that it points to.
+
+The following example defines a class named `Motorcycle` that contains five methods:
+
+[!CODE [csSnippets.Methods#40](../../samples/snippets/csharp/concepts/methods/methods40.cs#40)]
+
+Note that the `Motorcycle` class includes an overloaded method, `Drive`. Two methods have the same namne, but can be differentiated by their parameter types.
```

Comment:
typo "namne".
Suggestion: The second sentence could be: "Two methods can have the same name, but must be differentiated by their parameter types."